string filename
string mode
uint8 record_rate
